common/app/views/support/AmpAd.scala

```diff
@@ -33,16 +35,40 @@ case class AmpAdDataSlot(article: Article) {
   }
 }
 
-case class AmpAdRtcConfig() {
-  def toJsonString: String =
-    Json.obj("urls" -> (
-               Json.arr() ++
-                 (
-                   if(KruxSwitch.isSwitchedOn)
-                     // See https://trello.com/c/zBlwvOWI
-                     Json.arr("https://cdn.krxd.net/userdata/v2/amp/2196ddf0-947c-45ec-9b0d-0a82fb280cb8?segments_key=x&kuid_key=kuid")
-                   else Json.arr()
-                 )
-             )
-    ).toString()
+object AmpAdRtcConfig {
+
+  def toJsonString: String = {
+
+    val urls: Seq[(String, JsValueWrapper)] = {
+
+      def urlValue(url: String, switch: Switch): Option[String] =
```

Comment:
Would this function be built each time scala evaluates 'urls' or is scala clever enough to build it only once?

Reply:
It would be evaluated each time it's called.  It gets called twice.